code/HabitTracker/app/src/main/java/com/example/habittracker/DatabaseManager.java

```diff
@@ -242,19 +305,40 @@ public void onFailure(@NonNull Exception e) {
     }
 
     /**
-     * Updates a habit document
-     * @param prevTitle {@code String}                  The original title of the habit
-     * @param title     {@code String}                  The new title of the habit
-     * @param doc       {@code HashMap<String,Object>}  The newly updated document
+     * Modifies attribute of habit event for performing adding and editing operations
+     * @param userid        {String}    the current user
+     * @param habitTitle    {String}    the title of the selected habit
+     * @param he            {HabitEvent}the habit event to modify
+     * @param action        {int}       id of the type of action to perform in callback.
      */
-    public void updateHabitDocument(String userid, String prevTitle, String title, HashMap<String,Object> doc) {
-
-        DocumentReference colRef = usersColRef
+    public void getAndSetHabitId(String userid, String habitTitle, HabitEvent he, int action) {
+        // first query the habit
+        usersColRef
                 .document(userid)
                 .collection(habitsColName)
-                .document(prevTitle);
-
-        colRef.update(doc);
+                .whereEqualTo("title", habitTitle)
+                .get()
+                .addOnCompleteListener(new OnCompleteListener<QuerySnapshot>() {
+                    @Override
+                    public void onComplete(@NonNull Task<QuerySnapshot> task) {
+                        // successfully found a habit to edit
+                        if (task.isSuccessful()) {
+                            for (QueryDocumentSnapshot document : task.getResult()) {
+                                // set the habit id to the habit event
+                                he.setHabit(document.getReference().getId());
+
+                                if (action ==1) {
+                                    he.editDB();
```

Comment:
Could you leave a comment to explain why we need this change? I know you said a fake Habit would appear in database, but I don't understand why we need to updateDB/editDB here.

Reply:
Left the following:
>// handle the action here, these actions NEED to be performed thru a callback